modules/apps/content-dashboard/content-dashboard-journal-impl/src/main/java/com/liferay/content/dashboard/journal/internal/item/action/provider/PreviewJournalArticleContentDashboardItemVersionActionProvider.java

```diff
@@ -0,0 +1,117 @@
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.content.dashboard.journal.internal.item.action.provider;
+
+import com.liferay.asset.display.page.portlet.AssetDisplayPageFriendlyURLProvider;
+import com.liferay.content.dashboard.item.action.ContentDashboardItemVersionAction;
+import com.liferay.content.dashboard.item.action.provider.ContentDashboardItemVersionActionProvider;
+import com.liferay.content.dashboard.journal.internal.item.action.PreviewJournalArticleContentDashboardItemVersionAction;
+import com.liferay.journal.model.JournalArticle;
+import com.liferay.layout.display.page.LayoutDisplayPageProviderTracker;
+import com.liferay.layout.seo.kernel.LayoutSEOLinkManager;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.language.Language;
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.portlet.RequestBackedPortletURLFactoryUtil;
+import com.liferay.portal.kernel.security.permission.ActionKeys;
+import com.liferay.portal.kernel.security.permission.resource.ModelResourcePermission;
+import com.liferay.portal.kernel.service.LayoutLocalService;
+import com.liferay.portal.kernel.theme.ThemeDisplay;
+import com.liferay.portal.kernel.util.Portal;
+import com.liferay.portal.kernel.util.WebKeys;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Cristina González
+ */
+@Component(service = ContentDashboardItemVersionActionProvider.class)
+public class PreviewJournalArticleContentDashboardItemVersionActionProvider
+	implements ContentDashboardItemVersionActionProvider<JournalArticle> {
+
+	@Override
+	public ContentDashboardItemVersionAction
+		getContentDashboardItemVersionAction(
+			JournalArticle journalArticle,
+			HttpServletRequest httpServletRequest) {
+
+		if (!isShow(journalArticle, httpServletRequest)) {
+			return null;
+		}
+
+		return new PreviewJournalArticleContentDashboardItemVersionAction(
+			_assetDisplayPageFriendlyURLProvider, httpServletRequest,
+			journalArticle, _language, _layoutDisplayPageProviderTracker,
+			_layoutLocalService, _layoutSEOLinkManager, _portal,
+			RequestBackedPortletURLFactoryUtil.create(httpServletRequest));
+	}
+
+	@Override
+	public boolean isShow(
+		JournalArticle journalArticle, HttpServletRequest httpServletRequest) {
+
```

Comment:
Please I think that we have to add here the check when the article does not have a template and neither a display page template in this case we should not show the action.

Reply:
It is not needed, because we filter it in the Tracker.